README.md

```diff
@@ -10,268 +10,475 @@ hard-to-find defects early in the application development lifecycle.
+                returnStdout: true).trim()
+     env.GIT_COMMIT = sh(
+                script: 'git rev-parse HEAD',
+                returnStdout: true).trim()
+     ```
+   1. Add the following line after the `-project` line in the `env.CSONAR_CMD_ARGS` definition.
+     ```
+     -name "$GIT_BRANCH/$GIT_COMMIT/agent/$NODE_NAME/exe/$EXECUTOR_NUMBER/build/$BUILD_NUMBER" \
+     ```
 
-1.
 
-```
-cd /myfiles/src/projX
-rm -f *.o
-codesonar build /myfiles/csonar_projects/projX/${PROJNAME} -foreground  ${HUB} gcc -c A.c
-codesonar build /myfiles/csonar_projects/projX/${PROJNAME} -foreground ${HUB} gcc -c B.c
-codesonar build /myfiles/csonar_projects/projX/${PROJNAME} -foreground ${HUB} gcc -c C.c
-codesonar analyze /myfiles/csonar_projects/projX/${PROJNAME} -foreground ${HUB}
-```
+### C. Apply the CodeSonar plugin to your Jenkins Pipeline
 
-1. Collect the build step text into a single shell script /path/to/dir/mybuildscript.sh:
+Once your Jenkins Pipeline is correctly invoking the CodeSonar analysis, you can apply the CodeSonar plugin to collect
+analysis information from the hub.
 
-```
-cd /myfiles/src/projX
-rm -f *.o
-gcc -c A.c
-gcc -c B.c
-gcc -c C.c
+1. Go back to **Configure** page for your pipeline.
+1. Set up hub credentials to authorize the CodeSonar plugin's interactions with the hub. These should generally be for the same hub user account that you are using to authenticate your `codesonar build` and `codesonar analyze` commands, but you will need to set up separate parameters for the plugin to use.
+   | **Authentication Type** | **Parameter Name** | **Type** | **Value** |
+   |-------------------------|--------------------|--------------------|-----------|
+   | Username/Password       | `CSONAR_HUBUSERPASS` | Credentials Parameter | A **Username with password** containing the hub user account username and password |
+   | Certificate | `CSONAR_HUBUSERCERT` | Credentials Parameter | A **Certificate** containing the user certificate and its private key in PKCS#12 format.<p>**MANUAL:** How CodeSonar Works > CodeSonar Structure > Hub > Manually Generating and Uploading User Certificates</p> |
+1. [HTTPS hubs only] If your hub uses a self-signed server certificate, create a parameter to store a copy of this certificate so that the plugin can be instructed to trust it.
+   | **Parameter Name** | **Type** | **Value** |
+   |--------------------|--------------------|-----------|
+   | `CSONAR_HUB_CACERT_ID` | Credentials Parameter | A **secret file** containing the hub server certificate (PEM format). |
+1. Add the following step to your Pipeline after all your build steps.
+   ```
+   script {
+        def analysisId = sh(
+            script: 'cat "$CSONAR_WORKDIR/$CSONAR_PROJECT_NAME.prj_files/aid.txt"',
+            returnStdout: true).trim()
+        codesonar(
+            conditions: [],
+            credentialId: params.CSONAR_HUBUSERPASS,
+            hubAddress: params.CSONAR_HUB_ADDRESS,
+            socketTimeoutMS: 300000,
```

Comment:
socketTimeout is optional, and probably should not be included in the default configuration presented here.  We added that option to allow troubleshooting problems, but it is unlikely to be needed in practice.